app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt

```diff
@@ -236,7 +236,7 @@ constructor(
             .subscribeOn(backgroundScheduler)
             .observeOn(mainScheduler)
             .subscribeBy(
-                onNext  = { state = CourseView.State.CourseLoaded(it); continueLearning(); resolveCourseShareTooltip(it) },
+                onNext  = { removeCachedLesson(it) },
```

Comment:
Let make it in `doCompletableOnNext` or etc. Also it will make more sense to do it after enrollment action cause now we can execute it multiple times if we have multiple screens of the same course.